moto/iot/models.py

```diff
@@ -911,9 +911,17 @@ def _get_principal(self, principal_arn):
                 raise ResourceNotFoundException()
             principal = certs[0]
             return principal
-        else:
-            # TODO: search for cognito_ids
-            pass
+        from moto.cognitoidentity import cognitoidentity_backends
+
+        cognito = cognitoidentity_backends[self.region_name]
+        identities = []
+        for identity_pool in cognito.identity_pools:
+            pool_identities = cognito.pools_identities.get(identity_pool, None)
+            identities.extend(
+                [pi["IdentityId"] for pi in pool_identities.get("Identities", [])]
+            )
+        if principal_arn in identities:
+            return {"IdentityId": principal_arn}
```

Comment:
```suggestion
            identities = [pi["IdentityId"] for pi in pool_identities.get("Identities", [])]
            if principal_arn in identities:
                return {"IdentityId": principal_arn}
```
If I'm not mistaken, this would be a better solution in terms of performance. No need to cycle through all identity pools if the ID exists in the first one.

Reply:
I have already made the changes